addons/html_editor/static/src/core/delete_plugin.js

```diff
@@ -600,32 +600,29 @@ export class DeletePlugin extends Plugin {
     // If joined, the range is collapsed to start.
     // Returns the updated range.
     joinFragments(range) {
-        const { joinableLeft, joinableRight } = this.getJoinableFragments(range);
         const { commonAncestorContainer: commonAncestor } = range;
+        const { joinableLeft, joinableRight } = this.getJoinableFragments(range);
```

Comment:
Actually now that I think about it, we also have to take care not to put non-phrasing content inside paragraph-like elements, i.e. if a div can be merged with or in a p

Reply:
Sh*t. Indeed. The p + div scenario is not a problem right now because a div is always unmergeable... but that's purely by chance! And I can think of cases like p + li (containing blocks) in which things would go wrong.
Perhaps this should be handled by `joinBlocks`, something like:
-if left block allows only phrasing content, merge will only happen if every right block's child is phrasing content (other than the usual check for unmergeables).

Reply:
I'll write some tests and dig into it.